backend/ohq/views.py

```diff
@@ -607,3 +614,162 @@ class AnnouncementViewSet(viewsets.ModelViewSet):
+        if op["operationId"] == "listOccurrences":
+            op["parameters"].append(
+                {
+                    "name": "course",
+                    "in": "query",
+                    "required": True,
+                    "description": "A series of api/occurrences/?course=1&course=2 "
+                    + "- where the numbers are the course pks",
+                    "schema": {"type": "string"},
+                }
+            )
+            op["parameters"].append(
+                {
+                    "name": "filter_start",
+                    "in": "query",
+                    "required": True,
+                    "description": "The start date of the filter in ISO format in UTC+0.<br>"
+                    + "The returned events will have start_time strictly within "
+                    + "the range of the filter<br>"
+                    + "e.g 2021-10-05T12:41:37Z",
+                    "schema": {"type": "datetime"},
+                }
+            )
+            op["parameters"].append(
+                {
+                    "name": "filter_end",
+                    "in": "query",
+                    "required": True,
+                    "description": "The end date of the filter in ISO format in UTC+0<br>"
+                    + "e.g 2021-10-05T12:41:37Z",
+                    "schema": {"type": "datetime"},
+                }
+            )
+        return op
+
+
+class OccurrenceViewSet(
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    retrieve:
+    Return an Occurrence.
+
+    list:
+    You should pass in a list of course ids, along with the filter start and end dates,
+    and all the occurrences related to those courses will be returned to you.
+    Return a list of Occurrences.
+
+    update:
+    Update all fields in an Occurrence.
+    You must specify all of the fields or use a patch request.
+
+    partial_update:
+    Update certain fields in the Occurrece.
+    """
+
+    serializer_class = OccurrenceSerializer
+    permission_classes = [OccurrencePermission | IsSuperuser]
+    schema = OccurrenceSchema()
+
+    def list(self, request, *args, **kwargs):
+        # ensure timezone consitency
+        course_ids = request.GET.getlist("course")
+        filter_start = datetime.strptime(
+            request.GET.get("filter_start"), "%Y-%m-%dT%H:%M:%SZ"
+        ).replace(tzinfo=utc)
+        filter_end = datetime.strptime(request.GET.get("filter_end"), "%Y-%m-%dT%H:%M:%SZ").replace(
+            tzinfo=utc
+        )
+        courses = Course.objects.filter(pk__in=course_ids)
+        erm = EventRelationManager()
+        occurrences = []
+        for course in courses:
+            events_for_course = erm.get_events_for_object(course)
+            for event in events_for_course:
+                for occurrence in event.get_occurrences(filter_start, filter_end):
+                    occurrence.save()
```

Comment:
can you add a quick comment on why occurrence.save() needs to be called, for future reference